### PR TITLE
refactor: sqs out-of-process

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR completes the move of the sqs web server (read side) out-of-process into and into its own repository:
https://github.com/osmosis-labs/sqs

The write (ingester) side is refactor and simplified.

The read side part is already merged. Deployments are WIP.

## Testing and Verifying

Tested on a local node in conjunction with the web server itself.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: app/apptesting/cosmwasmpool.go [hunk 1]
```diff
@@ -22,20 +23,31 @@ const (
...
+// PrepareCustomTransmuterPool sets up a transmuter pool with the default parameters assuming that
+// the transmuter contract is stored under x/cosmwasmpool/bytecode in the Osmosis repository.
 func (s *KeeperTestHelper) PrepareCustomTransmuterPool(owner sdk.AccAddress, denoms []string) cosmwasmpooltypes.CosmWasmExtension {
+	return s.PrepareCustomTransmuterPoolCustomProject(owner, denoms, osmosisRepository, osmosisRepoTransmuterPath)
+}
+
+// PrepareCustomTransmuterPoolCustomProject sets up a transmuter pool with the custom parameters.
+// Gives flexibility for the helper to be reused outside of the Osmosis repository by providing custom
+// project name and path.
+func (s *KeeperTestHelper) PrepareCustomTransmuterPoolCustomProject(owner sdk.AccAddress, denoms []string, projectName, path string) cosmwasmpooltypes.CosmWasmExtension {
```
Comment (p0mvn) [inline app/apptesting/cosmwasmpool.go:45]: Note: these `apptesting` changes are necessary to allow reusing these helpers from another repository:
https://github.com/osmosis-labs/sqs

File: go.mod [hunk 2]
```diff
@@ -1,38 +1,36 @@
 module github.com/osmosis-labs/osmosis/v21
 
-go 1.20
+go 1.21
```
Comment (p0mvn) [inline go.mod:3]: AFAIK, there is no state-break between v1.20 and 1.21. Could someone please confirm?

I was getting warnings so updated this.

File: app/apptesting/cosmwasmpool.go [hunk 3]
```diff
@@ -22,20 +23,31 @@ const (
...
 // PrepareCosmWasmPool sets up a cosmwasm pool with the default parameters.
 func (s *KeeperTestHelper) PrepareCosmWasmPool() cosmwasmpooltypes.CosmWasmExtension {
 	return s.PrepareCustomTransmuterPool(s.TestAccs[0], []string{DefaultTransmuterDenomA, DefaultTransmuterDenomB})
 }
 
-// PrepareCustomTransmuterPool sets up a transmuter pool with the custom parameters.
+// PrepareCustomTransmuterPool sets up a transmuter pool with the default parameters assuming that
+// the transmuter contract is stored under x/cosmwasmpool/bytecode in the Osmosis repository.
 func (s *KeeperTestHelper) PrepareCustomTransmuterPool(owner sdk.AccAddress, denoms []string) cosmwasmpooltypes.CosmWasmExtension {
+	return s.PrepareCustomTransmuterPoolCustomProject(owner, denoms, osmosisRepository, osmosisRepoTransmuterPath)
```
Comment (p0mvn) [inline app/apptesting/cosmwasmpool.go:39]: Note: these apptesting changes are necessary to allow reusing these helpers from another repository:
https://github.com/osmosis-labs/sqs

File: go.mod (hunk 2 above)
Comment (p0mvn) [inline go.mod:3]: AFAIK, there is no state-break between v1.20 and 1.21. Could someone please confirm?

I was getting warnings so updated this.

Comment (czarcas7ic) [reply go.mod:3]: We should not back port if we change to v1.21, there have been historical examples of unexpected differences causing state incompatibilities

File: go.mod [hunk 4]
```diff
@@ -1,38 +1,36 @@
...
 
 require (
 	cosmossdk.io/api v0.3.1
 	cosmossdk.io/errors v1.0.0
 	cosmossdk.io/math v1.1.3-rc.1
 	cosmossdk.io/tools/rosetta v0.2.1
 	github.com/CosmWasm/wasmd v0.45.1-0.20231128163306-4b9b61faeaa3
-	github.com/cometbft/cometbft v0.37.2
+	github.com/cometbft/cometbft v0.38.0
```
Comment (czarcas7ic) [inline go.mod:11]: What is forcing this to bump? Both this and sdk import should not be forced to bump when running tidy

Comment (p0mvn) [reply go.mod:11]: They are bumped on tidy. Replaces are what matters anyway

File: go.mod (hunk 2 above)
Comment (p0mvn) [reply go.mod]: Changed back to 1.20

File: go.mod [hunk 5]
```diff
@@ -404,3 +405,7 @@ replace (
 
 // exclusion so we use v1.0.0
 exclude github.com/coinbase/rosetta-sdk-go v0.7.9
+
+exclude github.com/cosmos/cosmos-sdk v0.50.1
+
+exclude github.com/cometbft/cometbft v0.38.0
```
Comment (p0mvn) [inline go.mod:411]: These keep auto-updating on `go mod tidy`. That breaks proto-generation CI.

As a result, I manually excluded them here

File: cmd/osmosisd/cmd/root.go [hunk 6]
```diff
@@ -559,49 +559,6 @@ is-enabled = "false"
...
-# The height interval at which the candidate routes are recomputed and updated in
-# Redis
-route-update-height-interval = "{{ .SidecarQueryServerConfig.Router.RouteUpdateHeightInterval }}"
-
-# Whether to enable candidate route caching in Redis.
-route-cache-enabled = "{{ .SidecarQueryServerConfig.Router.RouteCacheEnabled }}"
-
-# The number of seconds to cache routes for before expiry.
-route-cache-expiry-seconds = "{{ .SidecarQueryServerConfig.Router.RouteCacheExpirySeconds }}"
-
```
Comment (czarcas7ic) [inline cmd/osmosisd/cmd/root.go:604]: Have we ever deprecated options in the config before? Will people with the old config error when starting the node? Can you please verify

Comment (p0mvn) [reply cmd/osmosisd/cmd/root.go:604]: If they have something present in the config that does not exist in the JSON struct we unmarshal to, that field will be silently ignored

**Review by czarcas7ic — APPROVED** [review]
Very high level ACK, there is a lot going on here but am fairly confident if there was an issue it would be confined to SQS

**Pull request merged**